webapp/src/main/java/com/box/l10n/mojito/service/machinetranslation/RepositoryMachineTranslationService.java

```diff
@@ -0,0 +1,127 @@
+    }
+
+    for (String targetBcp47Tag : targetBcp47Tags) {
+      final TextUnitSearcherParameters searchParameters = new TextUnitSearcherParameters();
+      searchParameters.setRepositoryIds(repository.getId());
+      searchParameters.setStatusFilter(StatusFilter.UNTRANSLATED);
+      searchParameters.setUsedFilter(UsedFilter.USED);
+      searchParameters.setLocaleTags(Arrays.asList(targetBcp47Tag));
+
+      final List<TextUnitDTO> textUnitDTOS = textUnitSearcher.search(searchParameters);
+
+      final List<String> sourceTexts =
+          textUnitDTOS.stream().map(TextUnitDTO::getSource).collect(Collectors.toList());
+
+      final TranslationsResponseDTO translationsResponseDTO =
+          machineTranslationService.getTranslations(
+              sourceTexts,
+              repository.getSourceLocale().getBcp47Tag(),
+              Arrays.asList(targetBcp47Tag),
+              false,
+              true,
+              null,
+              null);
+
+      final Map<String, List<TextUnitDTO>> sourceToTextUnitDTOs =
+          textUnitDTOS.stream().collect(Collectors.groupingBy(TextUnitDTO::getSource));
+
+      logger.info("Number of text unit to machine translate: {}", sourceToTextUnitDTOs.size());
+
+      if (sourceToTextUnitDTOs.size() > 300) {
+        throw new RuntimeException("Too many text unit, fail safe for now");
+      }
+
+      final List<TextUnitDTO> machineTranslatedTextUnitDTOs =
+          translationsResponseDTO.getTextUnitTranslations().stream()
+              .flatMap(
+                  textUnitTranslationGroupDTO -> {
+                    final String sourceText = textUnitTranslationGroupDTO.getSourceText();
+
+                    final Stream<TextUnitDTO> textUnitDTOStreamFor1Translation;
+
+                    if (!textUnitTranslationGroupDTO.getTranslations().isEmpty()) {
+                      // since we're doing 1 locale we just read the first entry
+                      final TranslationDTO translationDTO =
+                          textUnitTranslationGroupDTO.getTranslations().get(0);
+
+                      textUnitDTOStreamFor1Translation =
+                          sourceToTextUnitDTOs.getOrDefault(sourceText, Collections.emptyList())
+                              .stream()
+                              .map(
+                                  textUnitDTO -> {
+                                    textUnitDTO.setTarget(translationDTO.getText());
+                                    return textUnitDTO;
+                                  });
+                    } else {
+                      logger.error("There must be a translation available but if not, just skip");
+                      textUnitDTOStreamFor1Translation = Stream.empty();
+                    }
+
+                    return textUnitDTOStreamFor1Translation;
+                  })
+              .collect(Collectors.toList());
+
+      textUnitBatchImporterService.importTextUnits(machineTranslatedTextUnitDTOs, false, true);
```

Comment:
Shouldn't these translations be marked (imported) with a different status - so that someone looking at the repo can easily tell the difference between MT and human translated strings?

Reply:
Also - if we leverage across repos (I don't remember what happens for MT leveraging when repo names aren't provided - that might be the case) - MT translations could be returned back as actual translations - they poison our TMs essentially.